docs/auth/README.adoc

```diff
@@ -13,3 +15,149 @@ dependencies {
 
 == Usage
 
+To use the Auth SDK you'll first need to:
+
+* Import the Core module. See link:../core/README.adoc[the Core documentation].
+* Specify the schema of the redirect url for your Android project in `android.defaultConfig.manifestPlaceholders` of the apps `build.gradle` file.
+It should match the schema specified in the `Valid Redirect URL` section for the client in Keycloak.
+It is recommended to use the package name of the Android app as the schema of the redirect url to avoid conflicts.
+
+=== Initializing the module
+
+`AuthService` can be retrived using the `MobileCore#getInstance` method.
+
+[source,java]
+----
+AuthService authService = mobileCore.getInstance(AuthService.class);
+----
+
+Any subsequent `MobileCore#getInstance` method will return the same instance of `AuthService`.
+
+Before the `AuthService` can be used used `AuthService#init` must be invoked once in an app.
+
+If `AuthService#init` is not invoked then an `IllegalStateException` will be thrown when using the
+service.
+
+[source,java]
+----
+AuthServiceConfiguration authServiceConfig = new AuthServiceConfiguration.AuthConfigurationBuilder()
+    .withRedirectUri("org.aerogear.mobile.example:/callback")
+    .build();
+
+// Only invoke this once, every subsequent retrieval of the AuthService will retrieve the same, already initialized, instance.
+authService.init(context, authServiceConfig);
+----
+
+=== Retrieving the current user
+
+To retrieve the current authenticated user the `AuthService#currentUser` method can be invoked.
+
+[source,java]
+----
+// authService already initialized.
+UserPrincipal currentUser = authService.currentUser();
+
+if (currentUser) {
+    // User is already authenticated, perform an action
+} else {
+    // User is not authenticates, start authentication flow
+}
+----
+
+=== Authenticating
+
+To start the authentication invoke the `AuthService#login` method.
+
+[source,java]
+----
+// authService already initialized.
+AuthService authService = mobileCore.getInstance(AuthService.class);
```

Comment:
Hint. You probably want to avoid such extensive code snippets as this can be maintenance hell. Asciidocs allow you to reference and build snippets from actual code - so we can reference code in example application and it will work (sadly it's not working on github) 
Using links etc. will work as well. This way we will keep things updated all the time (as example app code is actually compiled). This was proven and tested in RainCatcher where we had rule to remove snippets longer than 10 lines

Reply:
I don't think these are intended to be snippets of the example app (though they currently are). So I don't think they're meant to keep in sync. The only times these should need to be updated is when the SDK API changes I would think. Though I agree a more automated approach would be good, I couldn't get the code snippet feature in AsciiDoc to function properly on GitHub.

Reply:
For now, I think embed those code inside this document is fine. Those code are meant to demo the API and they should relatively stable. If in the future we find these code snippets need updates a lot, we should then consider using referenced code instead.